test/functional/legacy/082_string_comparison_spec.lua

```diff
@@ -0,0 +1,118 @@
+-- Tests for case-insensitive UTF-8 comparisons (utf_strnicmp() in mbyte.c)
+-- Also test "g~ap".
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('case-insensitive string comparison in UTF-8', function()
+  setup(clear)
+
+  it('is working', function()
+    execute('set enc=utf8')
+    feed('ggdG<cr>')
+    -- first we define some functions that are needed for the test
```

Comment:
And this comment isn't entirely accurate because we both define functions *and* run the test cases inside `source`.